PowerArubaCP/Public/VM.ps1

```diff
@@ -0,0 +1,751 @@
+        [ipaddress]$ntp_primary,
+        [Parameter (Mandatory = $false)]
+        [ipaddress]$ntp_secondary,
+        [Parameter (Mandatory = $false)]
+        [switch]$fips
+    )
+
+    Begin {
+    }
+
+    Process {
+        if ($timezone_continent -xor $timezone_country) {
+            Throw "You need to specific Timezone Continent and Country on the sametime"
+        }
+        if (($data_ip -xor $data_netmask) -or ($data_netmask -xor $data_gateway)) {
+            Throw "You need to specific Data IP, netmask and Gateway on the sametime"
+        }
+        if (-not (Get-Command -name Set-VMKeystrokes -ErrorAction SilentlyContinue)) {
+            Throw "You need to install/import Set-VMKeystrokes script (Install-Script -Name VMKeystrokes)"
+        }
+
+        #Connection
+        Write-Output "Connection to console using default login/password"
+        Set-VMKeystrokes -VMName $name_vm -StringInput appadmin -ReturnCarriage $true 6>> $null
+        Start-Sleep 1
+        Set-VMKeystrokes -VMName $name_vm -StringInput eTIPS123 -ReturnCarriage $true 6>> $null
+
+        Start-Sleep 15
+
+        #Initial Setup
+        #Hostname
+        Write-Output "Configure hostname: $hostname"
+        Set-VMKeystrokes -VMName $name_vm -StringInput $hostname -ReturnCarriage $true 6>> $null
+        Start-Sleep 1
+
+        #Management IPv4
+        if ($version -eq "6.8") {
+            Write-Output "Configure Management IPv4: $mgmt_ip / $mgmt_netmask"
+            Set-VMKeystrokes -VMName $name_vm -StringInput $mgmt_ip.ToString() -ReturnCarriage $true 6>> $null
+            Start-Sleep 1
+            #Netmask
+            Set-VMKeystrokes -VMName $name_vm -StringInput (Convert-ArubaCPCIDR2Mask($mgmt_netmask)) -ReturnCarriage $true 6>> $null
+            Start-Sleep 1
+            #Gateway (Management)
+            Write-Output "Configure Management IPv4 Gateway: $mgmt_gateway"
+            Set-VMKeystrokes -VMName $name_vm -StringInput $mgmt_gateway.ToString() -ReturnCarriage $true 6>> $null
+            Start-Sleep 1
+
+        }
+        else {
+            #with 6.9.x using CIDR for netmask and add IPv6 support
+
+            $mgmt = $mgmt_ip.ToString() + "/" + $mgmt_netmask
+            Write-Output "Configure Management IPv4: $mgmt_ip / $mgmt_netmask"
+            Set-VMKeystrokes -VMName $name_vm -StringInput $mgmt -ReturnCarriage $true 6>> $null
+            Start-Sleep 1
+            Write-Output "Configure Management IPv4 Gateway: $mgmt_gateway"
+            Set-VMKeystrokes -VMName $name_vm -StringInput $mgmt_gateway.ToString() -ReturnCarriage $true 6>> $null
+            Start-Sleep 1
+
+            #Management IPv6 (Skip...)
+            Write-Output "Skip Configure Management IPv6..."
+            Set-VMKeystrokes -VMName $name_vm -SpecialKeyInput "KeyEnter" 6>> $null
+            Start-Sleep 1
+        }
+
+        #Data Port
+        if ($version -eq "6.8") {
+            if ($data_ip -and $data_netmask -and $data_gateway) {
```

Comment:
Data configuration is mandatory but here you check if they have values, is is usefull ?